src/core/BufferAttribute.js

```diff
@@ -199,15 +200,32 @@ Object.assign( BufferAttribute.prototype, {
 
 	applyMatrix3: function ( m ) {
 
-		for ( let i = 0, l = this.count; i < l; i ++ ) {
+		if ( this.itemSize === 2 ) {
 
-			_vector.x = this.getX( i );
-			_vector.y = this.getY( i );
-			_vector.z = this.getZ( i );
+			for ( let i = 0, l = this.count; i < l; i ++ ) {
 
-			_vector.applyMatrix3( m );
+				_vector2.x = this.getX( i );
+				_vector2.y = this.getY( i );
 
-			this.setXYZ( i, _vector.x, _vector.y, _vector.z );
+				_vector2.applyMatrix3( m );
+
+				this.setXY( i, _vector2.x, _vector2.y, );
+
+			}
+
+		} else {
```

Comment:
I would suggest this, instead... I think.
```js
} else if ( this.itemSize === 3 ) {
```

Reply:
Ok. Just a question: what should three do if `itemSize` differs from 2 and 3? Is this case possible?

Reply:
>what should three do if itemSize differs from 2 and 3?

Nothing for now.

>Is this case possible?

Yes, but a use case has not been presented.